File: CoinPay.Web/src/types/transaction.ts
export interface Transaction {
  id: number;
  transactionId?: string;
  amount: number;
  currency: string;
  type: string; // Payment, Refund, Transfer
  status: string; // Pending, Completed, Failed
  senderName?: string;
  receiverName?: string;
  description?: string;
  createdAt: string;
  completedAt?: string;
}

export enum TransactionStatus {
  Pending = 'pending',
  Processing = 'processing',
  Completed = 'completed',
  Failed = 'failed',
  Cancelled = 'cancelled',
}

export interface CreateTransactionRequest {
  amount: number;
  currency?: string;
  type?: string; // Payment, Refund, Transfer
  status?: string; // Pending, Completed, Failed
  senderName?: string;
  receiverName?: string;
  description?: string;
}

export interface TransactionResponse {
  transaction: Transaction;
  message?: string;
}

File: CoinPay.Api/Models/Transaction.cs
namespace CoinPay.Api.Models;

public class Transaction
{
    public int Id { get; set; }
    public string? TransactionId { get; set; }
    public decimal Amount { get; set; }
    public string Currency { get; set; } = "USD";
    public string Type { get; set; } = "Payment"; // Payment, Refund, Transfer
    public string Status { get; set; } = "Pending"; // Pending, Completed, Failed
    public string? SenderName { get; set; }
    public string? ReceiverName { get; set; }
    public string? Description { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? CompletedAt { get; set; }
}
